app/models/digital_object/publish_target.rb

```diff
@@ -147,8 +149,10 @@ def publish_digital_object(digital_object, do_ezid_update)
     if do_ezid_update && response.code == 200 && response.headers[:location].present?
       # By this point, all records should have an ezid. Let's update the status of
       # that ezid to :public, and send the latest published_object_url in case it changed.
-      unless digital_object.update_doi_metadata(response.headers[:location])
-        @errors.add(:ezid_response, "An error occurred while attempting to updated the ezid doi for this object.")
+      begin
+        digital_object.update_doi_metadata(response.headers[:location])
+      rescue DataciteErrorResponse, DataciteConnectionError, HyacinthError::DoiMissing => e
```

Comment:
I see you raising `Hyacinth::Exceptions::DataciteErrorResponse` and `Hyacinth::Exceptions::DataciteConnectionError` in one part of the code, but not catching those fully namespaced error classes here.  Is this a mistake?  If the tests are all passing, it seems like you may be missing some tests for this update.